### DISCUSSION TITLE
nvim-ts-autotag not working in my setup even after installing treesitter parsers for html and xml using TSInstall

### DISCUSSION THREAD

**Discussion by 231tr0n** [LazyVim/LazyVim#1805] [Q&A] [answered]
I am not able to get nvim-ts-autotag to work on the default starter template even after installing xml and html treesitter parsers using 
```
TSInstall html xml
```
Could someone please help me figure out what am I doing wrong and why is it not working for me. I just git cloned the starter repo and without any configuration and with treesitter parsers for html and xml installed I am still not able to get nvim-ts-autotag to work.

**Comment by dpetka2001** [comment]
I'm not sure about `xml`, but `html` is installed by default for Lazyvim. When i tried with a new starter template, i didn't have to do anything about `html`. I just opened a `.html` file and it worked out of the box. What does `:TSModuleInfo` say about html?

**Reply by 231tr0n** [reply]
![image](https://github.com/LazyVim/LazyVim/assets/56352048/c381b66f-252b-4c71-b5b3-c3482e210e2d)

**Reply by 231tr0n** [reply]
I did that and this is the output. Not sure what am I getting wrong.

**Reply by dpetka2001** [reply]
Did you change anything for the treesitter configuration? Do you have a link on github for your dotfiles?

**Reply by dpetka2001** [reply]
Did you install Neovim via snap?

**Reply by 231tr0n** [reply]
nope via pacman

**Reply by dpetka2001** [reply]
What version?

**Reply by 231tr0n** [reply]
I just installed the starter template and added these three options to the options.lua file.
```lua
vim.opt.undofile = false
vim.opt.expandtab = true
vim.opt.swapfile = true
```

**Reply by dpetka2001** [reply]
What version is your Neovim?

**Reply by 231tr0n** [reply]
NVIM v0.9.4
Build type: Release
LuaJIT 2.1.1696795921

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

**Reply by dpetka2001** [reply]
Does `:checkhealth nvim-treesitter` show any errors?

**Reply by 231tr0n** [reply]
Give me a sec i am fresh installing from scratch. Once done will run the command and give you the output

**Reply by 231tr0n** [reply]
![image](https://github.com/LazyVim/LazyVim/assets/56352048/643a2407-f2ff-4022-92fd-550328c72bcd)

**Reply by 231tr0n** [reply]
I did a fresh install and ensured html parser for treesitter is installed using TSInfo. Then I ran the command which you shared and this is the output

**Reply by dpetka2001** [reply, marked as answer]
Can you try the following? Create a file in `lua/plugins/treesitter.lua` with the following contents
```lua
return {
  {
    "nvim-treesitter/nvim-treesitter",
    dependencies = {
      { "windwp/nvim-ts-autotag" },
    },
    opts = {
      autotag = {
        enable = true,
      },
    },
  },
}
```
Can you test if this is working for you?

**Reply by 231tr0n** [reply]
Yes it started working now

**Reply by 231tr0n** [reply]
According to me from version 10 onwards this plugin should be enabled by default right. Was i supposed to create this config to work?

**Reply by dpetka2001** [reply]
I'm not sure. It has to do with the order the plugin is loaded. Pinging @folke to share his opinion. I also tried
```lua
  {
    "windwp/nvim-ts-autotag",
    event = function()
      return { "LazyFile", "VeryLazy" }
    end,
    opts = {},
  },
```
so it seems it doesn't have to be listed as a dependency to treesitter. Just maybe Folke consider changing the `event` the plugin is loading? Not sure, would appreciate any input.

**Reply by 231tr0n** [reply]
Anyways really thank you for listening to me and helping me out. I thought I was a goner and was almost about to switch back to lunarvim.

**Reply by dpetka2001** [reply]
No problem, glad it worked out for you in the end.

**Reply by folke** [reply]
You're right, it didn't work! Fixed
